### PR TITLE
Add tests for backend/

### PR DESCRIPTION
**Pull request by hawkowl**
### Which issue this PR addresses:

Further refactoring as a part of https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/6118707

### What this PR does / why we need it:

This makes the openshiftcluster Manager an Interface and adds direct tests for backend/.

### Test plan for issue:

It is tests :D

### Is there any documentation that needs to be updated for this PR?

N/A

### CODE REVIEW

**Review by mjudeikis — DISMISSED** [review]

File: pkg/backend/openshiftcluster_test.go [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+	expectedInstallPhase            api.InstallPhase
+}
+
+func TestBackendTry(t *testing.T) {
+	mockSubID := "00000000-0000-0000-0000-000000000000"
+	resourceID := fmt.Sprintf("/subscriptions/%s/resourcegroups/resourceGroup/providers/Microsoft.RedHatOpenShift/openShiftClusters/resourceName", mockSubID)
+
+	for _, tt := range []backendTestStruct{
+		{
+			name: "StateCreating with an InstallPhase set keeps it in Creating",
```
Comment (jim-minter) [inline pkg/backend/openshiftcluster_test.go:53]: can the first couple of test names be clarified at all?  Perhaps something like
"a queued doc in state Creating should stay in state Creating when the manager returns with properties.install set" ?

**Review by jim-minter — DISMISSED** [review]
Sorry for the delay in reviewing.  This looks pretty close.

File: test/util/statsd/statsd.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
+package statsd
```
[the remark this answers is not in the document]
Comment (hawkowl) [reply test/util/statsd/statsd.go:1]: done

File: pkg/backend/openshiftcluster_test.go (hunk 1 above)
Comment (hawkowl) [reply pkg/backend/openshiftcluster_test.go:53]: updated a few test names

File: pkg/backend/openshiftcluster_test.go [hunk 3]
```diff
@@ -0,0 +1,399 @@
...
+						inFlightDoc.OpenShiftCluster.Properties.Install = &api.Install{}
+						return nil
+					})
+					return err
+				})
+			},
+			expectedDocumentExists:    true,
+			expectedProvisioningState: api.ProvisioningStateCreating,
+			expectedInstallPhase:      api.InstallPhaseBootstrap,
+			expectedLogs: []map[string]types.GomegaMatcher{
```
[the remark this answers is not in the document]
Comment (hawkowl) [reply pkg/backend/openshiftcluster_test.go:84]: yeah, fair, done

File: pkg/backend/openshiftcluster_test.go [hunk 4]
```diff
@@ -0,0 +1,399 @@
...
+			}
+			if !worked {
+				t.Fatal("didnt do work")
+			}
+
+			// wait on the workers to finish their tasks
+			b.waitForWorkerCompletion()
+
+			out, err := dbOpenShiftClusters.Get(ctx, strings.ToLower(resourceID))
+			if tt.expectedDocumentExists {
```
[the remark this answers is not in the document]
Comment (hawkowl) [reply pkg/backend/openshiftcluster_test.go:358]: good idea! I was trying to maintain what was there, but I did write the checker code in the meantime :)

File: pkg/backend/openshiftcluster/openshiftcluster.go [hunk 5]
```diff
@@ -24,6 +26,15 @@ import (
 	"github.com/Azure/ARO-RP/pkg/util/subnet"
 )
 
+type ManagerInterface interface {
```
[the remark this answers is not in the document]
Comment (hawkowl) [reply pkg/backend/openshiftcluster/openshiftcluster.go:29]: done

**Review by mjudeikis — APPROVED** [review]

**Pull request merged**